datahub-web-react/src/app/entity/shared/containers/profile/header/EntityTabs.tsx

```diff
@@ -24,20 +25,19 @@ const Tab = styled(Tabs.TabPane)`
     line-height: 22px;
 `;
 
-export const EntityTabs = <T,>({ tabs, selectedTab }: Props) => {
+export const EntityTabs = <T,>({ loading, tabs, selectedTab }: Props) => {
```

Comment:
So what if we can keep the logic in here for routing only to visible tabs, but simply say that if the "selected tab name" is not in the "enabledAndVisibleTabs names", _then_ fall back to the first tab in "enabledAndVisibleTabs". 

If we do this, we may not need to check loading at all, since the enabledAndVisibleTabs will be re-computed each time the EntityData changes! 

How does that work?

Reply:
I could not test it yet, but I think this approach should technically work. In theory, there could be two potential problems. With this approach and without respecting the loading state we would do the routing twice: One routing takes place before loading the data and the other after loading the data.

From the browser's session history point of view this shouldn't be a problem, because when using useRouteToTab the state is replaced in the history and not appended. The only thing here is that a tab which will not be visible/accessible after loading the data might be visible for a short amount of time after the first routing (the tab itself will be visible because the EntityProfile is rendering the selected tab independently from the loading state of the entity, but before the routing takes place no tab is visible because the selected tab is undefined). However, I think this is something we can safely ignore.

The other problem is more theoretical, but I don't think that we should really ignore it: Assuming we have two tabs for an entity and the first tab is not visible before the loading of the data, but after and the second tab is visible before and after, then the first tab will not be the default tab, because the selected tab will not be undefined after the first routing and therefore the routing logic would not be applied anymore. The question here is whether this is fine or if we want to intentionally delay the routing here because of this. If we want to delay it, we would need the loading state in EntityTabs. Maybe we should then also render a placeholder as long the selected tab is undefined or alternatively to be consistent with EntityTabs (when the selected tab is undefined the first tab seems to be automatically selected by the component, see also here: https://github.com/ant-design/ant-design/issues/39730) render always the first visible tab in EntityProfile (but not route to this tab in this case, to be able to apply the "show the first visible and enabled tab as default after loading"-logic).

If you are agree that we should respect the loading state and delay the routing in EntityTabs (I don't think there is another solution for this?), I will try to prepare something by moving the loading state into the EntityContext (even if we currently only need the loading state in EntityTabs it could also be useful in the tabs themselves at one place or another).